modules/gdscript/gdscript_analyzer.cpp

```diff
@@ -3192,8 +3192,7 @@ void GDScriptAnalyzer::reduce_preload(GDScriptParser::PreloadNode *p_preload) {
 		if (!FileAccess::exists(p_preload->resolved_path)) {
 			push_error(vformat(R"(Preload file "%s" does not exist.)", p_preload->resolved_path), p_preload->path);
 		} else {
-			// TODO: Don't load if validating: use completion cache.
-			p_preload->resource = ResourceLoader::load(p_preload->resolved_path);
+			p_preload->resource = GDScriptCache::get_shallow_script(p_preload->resolved_path, parser->script_path);
```

Comment:
I know that I need to add a check to see if the preloaded path is a script or not before using GDScriptCache.